akka-actor-tests/src/test/scala/akka/actor/ActorSystemDispatchersSpec.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor
+
+import akka.actor.setup.ActorSystemSetup
+import akka.dispatch.ExecutionContexts
+import akka.testkit.{ AkkaSpec, ImplicitSender, TestActors, TestProbe }
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+object ActorSystemDispatchersSpec {
+
+  class SnitchingExecutionContext(testActor: ActorRef, underlying: ExecutionContext) extends ExecutionContext {
+
+    def execute(runnable: Runnable): Unit = {
+      testActor ! "called"
+      underlying.execute(runnable)
+    }
+
+    def reportFailure(t: Throwable): Unit = {
+      testActor ! "failed"
+      underlying.reportFailure(t)
+    }
+  }
+
+}
+
+class ActorSystemDispatchersSpec extends AkkaSpec with ImplicitSender {
+
+  import ActorSystemDispatchersSpec._
+
+  "The ActorSystem" must {
+
+    "work with a passed in ExecutionContext" in {
+      val ecProbe = TestProbe()
+      val ec = new SnitchingExecutionContext(ecProbe.ref, ExecutionContexts.global())
+
+      val system2 = ActorSystem(name = "ActorSystemDispatchersSpec-passed-in-ec", defaultExecutionContext = Some(ec))
+
+      try {
+        val ref = system2.actorOf(Props(new Actor {
+          def receive = {
+            case "ping" => sender() ! "pong"
+          }
+        }))
+
+        val probe = TestProbe()
+
+        ref.tell("ping", probe.ref)
+
+        ecProbe.expectMsg(1.second, "called")
+        probe.expectMsg(1.second, "pong")
+      } finally {
+        shutdown(system2)
+      }
+    }
+
+    "not use passed in ExecutionContext if executor is configured" in {
+      val ecProbe = TestProbe()
+      val ec = new SnitchingExecutionContext(ecProbe.ref, ExecutionContexts.global())
+
+      val config = ConfigFactory.parseString("akka.actor.default-dispatcher.executor = \"fork-join-executor\"")
+      val system2 = ActorSystem(
+        name = "ActorSystemDispatchersSpec-ec-configured",
+        config = Some(config),
+        defaultExecutionContext = Some(ec))
+
+      try {
+        val ref = system2.actorOf(TestActors.echoActorProps)
+        val probe = TestProbe()
+
+        ref.tell("ping", probe.ref)
+
+        ecProbe.expectNoMessage(200.millis)
+        probe.expectMsg(1.second, "ping")
+      } finally {
+        shutdown(system2)
+      }
+    }
```

Comment:
These two I just moved from ActorSystemSpec to keep dispatcher related things together a bit